src/passes/ReorderGlobals.cpp

```diff
@@ -77,7 +77,7 @@ struct ReorderGlobals : public Pass {
   // use the index of the global in the original ordering to identify each
   // global. A different ordering is then a vector of old indices, saying where
   // each element comes from, which is logically a mapping between indices.
-  using IndexIndexMap = std::vector<Index>;
+  using IndexIndexMap = std::vector<size_t>;
```

Comment:
Can this not stay `Index`? We know wasm sizes fit in 32 bits and that may help us quite a bit with cache locality.

Reply:
Not unless we want to convert to `size_t` before calling the topo sort utils or if we convert all the topo sort utils to use `Index`. The latter is probably a good idea, but would be a larger change. Let's consider that as a follow-up.